app/controllers/concerns/authorization.rb

```diff
@@ -9,6 +9,6 @@ module Authorization
   private
 
   def user_not_authorized
-    redirect_to(root_path)
+    redirect_to new_user_session_path, alert: I18n.t("flash.user_not_authorized")
```

Comment:
I think that you confuse authentication and authorization concepts. We should redirect to `new_user_session_path` when a user is not logged in and to root_path when he is not able to perform some action (with a relevant flash error message).